common/lib/capa/capa/capa_problem.py

```diff
@@ -191,6 +193,48 @@ def __init__(self, problem_text, id, capa_system, state=None, seed=None):
 
         self.extracted_tree = self._extract_html(self.tree)
 
+    def make_xml_compatible(self, tree):
+        """
+        Adjust tree xml in-place for compatibility, such as
+        supporting an old xml format by translating it to a new format.
```

Comment:
This seems a little scary, the notion that there's some arbitrary "old" and arbitrary "new" formats.

This perhaps raises the specter of versioning the XML....